src/java/us/kbase/mobu/shell_launcher/ShellLauncher.java

```diff
@@ -0,0 +1,48 @@
+package us.kbase.mobu.shell_launcher;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.InterruptedException;
+import java.lang.ProcessBuilder;
+import java.lang.Process;
+import java.util.Map;
+import java.nio.file.Paths;
+
+import org.yaml.snakeyaml.Yaml;
+
+import us.kbase.mobu.util.DirUtils;
+import us.kbase.mobu.util.TextUtils;
+import us.kbase.mobu.ModuleBuilder;
+
+public class ShellLauncher {
+    File moduleDir;
+    String moduleName;
+    String workDir;
+
+    public ShellLauncher() throws IOException {
+        moduleDir = DirUtils.findModuleDir();
```

Comment:
It seems like all this stuff is something that's going to happen over and over - getting the module dir, module name, and work dir. Maybe make a class that does all that stuff in one place?